### DISCUSSION TITLE
How to Use a Fine-Tuned CRAFT Model in EasyOCR

### DISCUSSION THREAD

**Discussion by tomatoboy-hub** [JaidedAI/EasyOCR#1205] [Q&A]
Hello,

I have successfully fine-tuned a CRAFT model using the training script provided in the EasyOCR's CRAFT training documentation (link: https://github.com/JaidedAI/EasyOCR/blob/master/trainer/craft/README.md), and obtained a file named `CRAFT_clr_amp_48.pth`. However, I am unsure how to integrate this fine-tuned model into EasyOCR for text detection purposes.

Could you please provide guidance or examples on how to use a fine-tuned `.pth` model with EasyOCR for text detection? 

Any help or pointers would be greatly appreciated.

Thank you.

**Comment by rezakarbasi** [comment]
The model's path is set here:
[getDetectorPath](https://github.com/JaidedAI/EasyOCR/blob/c999505ef6b43be1c4ee36aa04ad979175178352/easyocr/easyocr.py#L235C31-L235C45)

Which means that you either need to put the file in one of the lib's default paths, or register a new type of models in the specified file

**Comment by donjuanpond** [comment]
You can try using the code from this tutorial: https://eivindkjosbakken.wordpress.com/2024/01/31/how-to-fine-tune-the-craft-model-in-easyocr/. It loads the model into EasyOCR's Reader class as a custom detector. However, I don't know how to use this model for anything other than EasyOCR's interface.
